### ISSUE TITLE
Inform user that loading has failed when dependency fetch fails

### ISSUE THREAD

**Issue by Mpstark** [BattletechModders/ModTek#101]
The most common fail state of mods that causes an infinite load is that something from `DataManager` was requested and wasn't present, generally in this `RequestDependency` method. The return value, indicating initial success or failure, is almost always ignored, leading to the infinite loading issue. While a perfect solution kicks back to the main menu with a popup (much like the failed to load save), this seems to work almost as well.

I've tested the following code briefly and it works for trivial tests. More testing would be necessary if we were to want to include this. There are other methods of `DataManager` that could fail in a similar way, but I believe that this is the one most likely to fail.

```c#
using BattleTech.Data;
using BattleTech.UI;
using Harmony;

namespace ModTek
{
    /// <summary>
    /// Patch the RequestDependency to change LoadingCurtain to a fail state if it returns that it cannot load a dependency
    /// Infinite loads are a horrible fail-state, this informs the user that something has gone wrong
    /// </summary>
    [HarmonyPatch(typeof(DataManager), "RequestDependency")]
    public static class DataManager_RequestDependency_Patch
    {
        public static void Postfix(bool __result)
        {
            if (__result == false && LoadingCurtain.IsVisible)
            {
                LoadingCurtain.ShowPopup("Loading Failed -- Check output_log.txt!");
            }
        }
    }
}
```

@m22spencer: Do you have any additional insight to this sort of thing?

This is what it looks like when it trips:

![loadingfailed](https://user-images.githubusercontent.com/5605487/51176609-fe228100-1882-11e9-8f5b-d95ef849aab6.png)

**Comment by Mpstark** [comment]
I suppose that the message could be more clear that a dependency is missing.

**Comment by m22spencer** [comment]
I've tried a similar patch in the past here: https://github.com/m22spencer/BattletechPerformanceFix/blob/master/source/MissingAssetsContinueLoad.cs#L52

It works off the asset manifest instead of the result.  I can tell you that it most definitely does not work with RT (and some other mods).  There are some situations where they need assets to be missing apparently.

Since your patch doesn't block (or "corrupt" by substituting items), it's probably fine. May lead to some people getting the message on a valid load though.

**Comment by janxious** [comment]
might be reasonable to check for the presence of FYLS and change the message to cleaned_output_log.txt.

Does this message appear when running an otherwise vanilla setup with just BTML and ModTek installed? Just curious if this trips on the base game and might cause false positives as @m22spencer has suggested.

**Comment by Mpstark** [comment]
@m22spencer That's very helpful! Unfortunately, I believe that this as written currently will prevent the `LoadingCurtain` from coming down in many cases. I've got some ideas on how to improve this.

@janxious: I don't think that it's common in vanilla, but I could be wrong. As to logging, the base game already spits out a nice log message that is copied over to FYLS -- I don't see a particular reason to change the message tbh.

**Comment by CptMoore** [comment]
Regarding FYLS, id just make the whole thing configurable by mod author, so RT can even say go to discordchannel and post cleaned...

Regarding the check, i definatly know that lots of code does not really care if something is loaded or not, there still is a vanilla issue where some items in inventory are not loaded when you first get into the mechlab, and you have to reenter the lab for them to load. However for jsons, if they were properly requested in dm, and then failed to load in dm, that definatly always stopped working.

I wounder if we couldnt just resolve those jsons issues by injecting dummy objects?

In any case, knowing when a load failed is already a huge win.

**Comment by Mpstark** [comment]
@CptMoore I know @m22spencer tried something like that in his linked post.

I've got a pull request #103 that addresses this.

**Comment by Mpstark** [comment]
Merged with #103.
